### PR TITLE
FAT-999: When an existing loan is claimed returned, update claimedReturnedDate, suspend any lost item fees billed and refund any lost item fees paid

### PR DESCRIPTION
**Pull request by yusuf-murodov-epam**
## Purpose
[FAT-999](https://issues.folio.org/browse/FAT-999)
_When an existing loan is claimed returned, update claimedReturnedDate, suspend any lost item fees billed and refund any lost item fees paid_

## Approach
- Reproduced test scenario with UI.
- Learned necessary sent requests and then necessary received responses 
- Based on the request/response created karate test scenario

## Learning
- learn karate framework from original doc file on the github
- looked other scenarios how they are described.

### CODE REVIEW

File: mod-circulation/src/main/resources/vega/mod-circulation/features/samples/claim-item-returned-entity-request.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "id": "#(claimItemReturnedId)",
+  "comment": "claim item returned",
+  "itemClaimedReturnedDateTime": "#(claimItemReturnedDate)"
+}
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/samples/claim-item-returned-entity-request.json:5]: add empty line at the end

File: mod-circulation/src/main/resources/vega/mod-circulation/features/samples/feefine/refund-to-patron-entity-request.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": "#(refundToPatronId)",
+  "accountId": "#(accountId)",
+  "amountAction": "#(amount)",
+  "balance": "#(balance)",
+  "typeAction": "#(typeAction)",
+  "userId": "#(userId)",
+  "createdAt": "#(servicePointId)"
+}
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/samples/feefine/refund-to-patron-entity-request.json:9]: add empty line at the end

File: mod-circulation/src/main/resources/vega/mod-circulation/features/samples/update-account-entity-request.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "ownerId": "#(ownerId)",
+  "paymentStatus": {
+    "name": "Suspended claim returned"
+  },
+  "remaining": "#(amount)",
+  "status": {
+    "name": "Open"
+  },
+  "userId": "#(userId)"
+}
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/samples/update-account-entity-request.json:23]: add empty line at the end

File: mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature [hunk 4]
```diff
@@ -690,4 +690,108 @@ Feature: Loans tests
...
+
+    # pay the lost item fine and verify that the fine is paid fully
+    * def postPayResponse = call read('classpath:vega/mod-circulation/features/util/initData.feature@PostPay') { accountId: #(lostItemFineAccountId), amount: #(lostItemFineAmount) }
+    * def feefineactionInResponse = postPayResponse.response.feefineactions[0]
+    And match feefineactionInResponse.typeAction == 'Paid fully'
+    And match feefineactionInResponse.amountAction == lostItemFineAmount
+    And match feefineactionInResponse.accountId == lostItemFineAccountId
+
+    # claim the item as returned
+    * def claimItemReturnedDate = '2021-12-07T20:30:46.000Z'
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature:762]:
```suggestion
    * def claimItemReturnedDate = call read('classpath:vega/mod-circulation/features/util/get-time-now-function.js')
```

File: mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature [hunk 5]
```diff
@@ -690,4 +690,108 @@ Feature: Loans tests
...
+    And match refundedFeeInResponse.response.balance == lostItemFineAmount
+
+    # verify that the fine billed is suspended
+    Given path 'accounts'
+    And param query = '(loanId==' + loanId + ')'
+    When method GET
+    Then status 200
+    * def accounts = response.accounts;
+    And match accounts[0].paymentStatus.name == 'Suspended claim returned'
+    And match accounts[1].paymentStatus.name == 'Suspended claim returned'
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature:797]: add empty line at the end

File: mod-circulation/src/main/resources/vega/mod-circulation/features/util/initData.feature [hunk 6]
```diff
@@ -356,3 +356,33 @@ Feature: init data for mod-circulation
...
+
+    Given path 'circulation/loans/' + loanId + '/claim-item-returned'
+    And request claimItemReturnedRequest
+    When method POST
+    Then status 204
+
+  @PostRefundFee
+  Scenario: refund fee to patron
+    * def refundFeeId = call uuid1
+
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/util/initData.feature:373]: remove empty line

File: mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature [hunk 7]
```diff
@@ -690,4 +690,108 @@ Feature: Loans tests
...
+    # update accounts
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PutAccount') { accountId: #(lostItemFineAccountId), barcode: #(extItemBarcode), amount: #(lostItemFineAmount), feeFineId: #(lostItemFeeFineId), feeFineType: #(lostItemFeeFineType), holdingsRecordId: #(holdingId) }
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PutAccount') { accountId: #(lostItemProcessingFineAccountId), barcode: #(extItemBarcode), amount: #(lostItemProcessingFineAmount), feeFineId: #(lostItemProcessingFeeFineId), feeFineType: #(lostItemProcessingFeeFineType), holdingsRecordId: #(holdingId) }
+
+    # verify the loaned item status is claimed-returned
+    Given path 'circulation/loans/' + loanId
+    When method GET
+    Then status 200
+    * def claimReturnedItemInResponse = response.item
+    And match claimReturnedItemInResponse.status.name == 'Claimed returned'
```
Comment (viktor-draban) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature:774]:
```suggestion
    And match $.item.status.name == 'Claimed returned'
```

File: acquisitions/src/main/resources/domain/mod-finance/features/Create-planned-budget-without-expense-classes-when-there-is-no-current-budget.feature [hunk 8]
```diff
@@ -1,38 +0,0 @@
...
-  Scenario: Check, that planned budget doesn't have expense classes and created for planned current fiscal year. If current budget absent
-    Given path 'finance/budgets', plannedBudgetWithoutExpenseClassesId2
-    When method GET
-    Then status 200
-    And match $.statusExpenseClasses == '#[0]'
-    And match $.fundId == '#(fundWithoutCurrBudgetId)'
-    And match $.fiscalYearId == '#(globalPlannedFiscalYearId)'
-
-
-
```
Comment (viktor-draban) [inline acquisitions/src/main/resources/domain/mod-finance/features/Create-planned-budget-without-expense-classes-when-there-is-no-current-budget.feature:38]: why ?

File: acquisitions/src/main/resources/domain/mod-finance/features/When-creating-budget-add-expense-classes-from-previous-budget-automatically.feature [hunk 9]
```diff
@@ -1,50 +0,0 @@
...
-    When method GET
-    Then status 200
-    * def statusExpenseClass = $.statusExpenseClasses[0]
-    And match $.statusExpenseClasses == '#[1]'
-    And match statusExpenseClass.expenseClassId == '#(globalElecExpenseClassId)'
-    And match $.fundId == '#(fundWithCurrBudgetId1)'
-    And match $.fiscalYearId == '#(globalPlannedFiscalYearId)'
-
-
-
```
Comment (viktor-draban) [inline acquisitions/src/main/resources/domain/mod-finance/features/When-creating-budget-add-expense-classes-from-previous-budget-automatically.feature:50]: why ?

File: acquisitions/src/main/resources/domain/mod-finance/features/When-creating-budget-add-expense-classes-if-them-provided-by-user.feature [hunk 10]
```diff
@@ -1,46 +0,0 @@
...
-
-  Scenario: Check, that planned budget has expense classes which were provided in creation time
-    Given path 'finance/budgets', plannedBudgetWithExpenseClassesId
-    When method GET
-    Then status 200
-    * def statusExpenseClass = $.statusExpenseClasses[0]
-    And match $.statusExpenseClasses == '#[1]'
-    And match statusExpenseClass.expenseClassId == '#(globalPrnExpenseClassId)'
-    And match $.fundId == '#(fundWithCurrBudgetId)'
-    And match $.fiscalYearId == '#(globalPlannedFiscalYearId)'
```
Comment (viktor-draban) [inline acquisitions/src/main/resources/domain/mod-finance/features/When-creating-budget-add-expense-classes-if-them-provided-by-user.feature:46]: why ?

File: acquisitions/src/main/resources/domain/mod-finance/features/Create-planned-budget-without-expense-classes-when-there-is-no-current-budget.feature (hunk 8 above)
Comment (yusuf-murodov-epam) [reply acquisitions/src/main/resources/domain/mod-finance/features/Create-planned-budget-without-expense-classes-when-there-is-no-current-budget.feature:38]: this file is moved to another package, and when i pushed my changes, these red lines occurred in my PR

Comment (viktor-draban) [reply acquisitions/src/main/resources/domain/mod-finance/features/Create-planned-budget-without-expense-classes-when-there-is-no-current-budget.feature:38]: anyway these changes mustn't be in your PR

**Pull request closed without merging**